src/config/kubeconfig.js

```diff
@@ -1,9 +1,73 @@
 'use strict';
 
-const process = require('process');
 const pth = require('path');
 const fs = require('fs');
-const k8s = require('@kubernetes/client-node');
+// TODO: Uncomment after PR https://github.com/kubernetes-client/javascript/pull/748 lands
+// const k8s = require('@kubernetes/client-node');
+
+// TODO: Delete next two functions
+// after PR https://github.com/kubernetes-client/javascript/pull/748 lands
+
+function dropDuplicatesAndNils(a) {
+  return a.reduce((acceptedValues, currentValue) => {
+    // Good-enough algorithm for reducing a small (3 items at this point) array into an ordered list
+    // of unique non-empty strings.
+    if (currentValue && !acceptedValues.includes(currentValue)) {
+      return acceptedValues.concat(currentValue);
+    } else {
+      return acceptedValues;
+    }
+  }, []);
+}
+
+// Only public for testing.
+function findHomeDir() {
```

Comment:
We also have a [`findHome()`](https://github.com/rancher-sandbox/rancher-desktop/blob/6c3026b7f122af28a3b655dde91b4029aff87240/src/k8s-engine/k3sHelper.ts#L511) elsewhere that claims to be compatible to this; we should unify the code (probably just by exporting this one and junking that one).  At a minimum, that needs to be fixed to use the same logic.

Reply:
Yes, good idea. Forgot about that one.